resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -241,15 +248,23 @@ private[yarn] class ExecutorRunnable(
       }
     }
 
-    // Add log urls
+    // Add log urls, as well as executor attributes
     container.foreach { c =>
+      val containerId = ConverterUtils.toString(c.getId)
+      val address = c.getNodeHttpAddress
+
+      env("SPARK_EXECUTOR_ATTRIBUTE_HTTP_SCHEME") = httpScheme
```

Comment:
The way these attributes are being propagated is a bit weird, and limiting in the long run. The driver needs to be able to figure them out, and the executor is just piping the env variables back to the driver.

I don't really have a suggestion for a different way, but just wanted to point out that this is a bit weird. e.g. in YARN the executor can figure out all this info by itself; there just isn't a programmatic interface for it to send that information to the driver as part of the registration message.

Reply:
Yeah I followed the way how current log URLs are propagated, and the code was aged for years so I simply guessed we may haven't been able to find better alternatives.

I agreed we may be also able to figure out these information in executor side, because we are doing the same thing from driver in cluster mode which container is already launched. One thing I'm afraid for making change is, current approach can be leveraged for all of resource managers (a general solution), whereas we may need to have (slightly) different implementation of `CoarseGrainedExecutorBackend` once we take the case for YARN. 

If we are OK to try tackling on this, overriding `extractLogUrls` and `extractAttributes` seems the simplest way to achieve it. We might change `CoarseGrainedExecutorBackend` to become abstract class to make clear two methods should be open to override, but it's OK if we simply extend `CoarseGrainedExecutorBackend` and have YARN's own executor backend which only overrides two above methods.

Would love to hear your thought about this.